Cargo.toml

```diff
@@ -0,0 +1,16 @@
+[package]
+name = "vsock-relay"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
```

Comment:
Given this controls the trusted connection between the shim and the enclave, I'm thinking it would be useful to version lock everything here for greater stability. Wdyt?

Reply:
also as mentioned in the sec review issue, please add a cargo audit github action

Reply:
Yeah specify full version and using the `^` to lock it is what I had in mind. Then we can use renovate bot to bump them as we want to or when the `cargo audit` flags an issue with a dependency for us. The thinking here is that random version bumps may introduce odd side effects or open up the possibility of supply chain attacks. So locking makes it harder to encounter these issues and then use automation to keep us up to date still.

Reply:
Sorry, still having trouble understanding what you're asking for. To summarize my understanding:

The `^` sigil [sets a minimum version](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements) but still allows updates, the same as a bare version number. So this doesn't prevent side-effects from automatic updates and supply-chain attacks.

There are afiak, two ways to enforce specific versions:
1. Use the `=` sigil to specify and exact version in `Cargo.toml`.
2. Rely on the versions recorded in `Cargo.lock`.

With (1) any version change to a direct dependency involves an edit to Cargo.toml. Dependabot/Renovate can automate this. However, it does nothing  to control the version of transitive dependencies, which are still taken from `Cargo.lock`.

With (2) a local build will use the exact recorded versions, so developers and CI will use those versions, and be able to review changes, as long as `Cargo.lock` is tracked under version control. Dependabot/Renovate can automate updates to just `Cargo.lock` if `Cargo.toml` doesn't specify a minimum tiny revision. However, one has to be careful to `cargo install --locked` to avoid upgrades when building outside the context of a source checkout. 

My preferred practice for binary crates like this is to specify a *minimum* version number of `Cargo.toml` and rely on `Cargo.lock` to track review of updates. This is slightly lower-friction for devs as `cargo update` works and reviews have one less file to check. Library crates are a different story since any `Cargo.lock` will be ignored in downstream builds; for those I like to have a three-part minimum in `Cargo.toml` so there's a place to record reviews of updates, but downstream builders aren't blocked from updating to a newer version; bug fixes being more common than supply-chain attacks so far.

So are you asking for  "=x.y.z" primary dependency references in `Cargo.toml` or is `Cargo.lock` ok?

Reply:
just fyi, the version locking approach that we use in this repo is the same approach that we use in our other rust projects

Reply:
![Screenshot 2023-07-20 at 5 33 47 PM](https://github.com/brave-experiments/vsock-relay/assets/23125059/04572b8e-c2f8-4dc0-ac3d-3321c69f9249)

I'm seeing that as the behavior of `~` not `^`. In any case, I can tell you understand what I'm asking for, so we can sort if we decide it's worth doing precise versioning.

I was asking for primary dependency locking in `Cargo.toml` (with whichever syntax it is for rust-lang) because I was thinking that this would keep transitive dependencies in `Cargo.lock` from being updated too. Keeping dependencies more stable seems to be the more secure method which is what I was aiming for with these changes. However, it sounds like I might be missing some tradeoffs that come with this change.

 > My preferred practice for binary crates like this is to specify a minimum version number in Cargo.toml and rely on Cargo.lock to track review of updates. This is slightly lower-friction for devs as cargo update works and reviews have one less file to check. Library crates are a different story since any Cargo.lock will be ignored in downstream builds; for those I like to have a three-part minimum in Cargo.toml so there's a place to record reviews of updates, but downstream builders aren't blocked from updating to a newer version; bug fixes being more common than supply-chain attacks so far.

I don't think I'm fully understanding the tradeoffs here in this case, so let me ask a few questions to make sure I'm not missing anything.

1. Does `cargo update` break for both developers of this library and downstream devs if we version lock?
2. Would we expect that when utilizing this package downstream, that the transient dependencies from this could be bumped even if left untouched within this package?
3. Would that still happen downstream with the transient dependencies if the versions get locked here too?

What I was hoping to prevent was really accidental transient version bumping, both within this repo and for downstream, so that we could have more stability for the behavior of this library because it's enforcing the trust boundary between the shim and the enclave. If this is too burdensome though (which it sounds like it may be) then it might not be worth it.